### ISSUE TITLE
Set k8sServiceHost and k8sServicePort with kubeproxy-free mode

### ISSUE THREAD

**Issue by xcloudx87** [cilium/cilium-cli#731]
## Proposal / RFE
Able to set k8sServiceHost and k8sServicePort with kubeproxy-free mode.

**Is your feature request related to a problem?**
I saw that we had feature to disable kubeproxy-free but didn't see option to set k8sServiceHost and k8sServicePort  when using cilium CLI. How to get it work with CLI? Otherwise we will need to use helm to install Cilium without CLI.

**Describe the solution you'd like**
Have option to set k8sServiceHost and k8sServicePort with from install OPTION.

**Comment by tklauser** [comment]
Related to #243

**Comment by vosdev** [comment]
I tried to work around this problem by setting the `k8s-api-server` `https://ebpf-test.example.com:6443` in the configmap, thinking it would yield the same result as setting `k8sServiceHost` and `k8sServicePort`.

It helped to get my cilium pods running but everything else still does not function. Network between pods does not work. All connectivity tests from `cilium connectivity test` fail. coredns cannot talk to kube-api-server etc..

**Comment by tklauser** [comment]
This will be solved by #717 which will switch cilium-cli to use helm charts for installation and allows overriding arbitrary helm options such as k8sServiceHost and k8sServicePort mentioned in this issue.

**Comment by vincentmli** [comment]
@tklauser I ran into same issue, wonder if this improvement has been released, I tried v0.10.5, not sure how to use the option :)

**Comment by vosdev** [comment]
@vincentmli it can be configured through the new --set-helm-string option like such:
```
cilium install --version=v1.11.4 --kube-proxy-replacement strict --helm-set-string=k8sServiceHost=k8s-api.example.com,k8sServicePort=6443,ipv6.enabled=true,ipam.mode=kubernetes,enableIPv6Masquerade=false,externalIPs=true
```

I believe this issue can be closed as https://github.com/cilium/cilium-cli/pull/717 has been merged some time ago.

**Comment by vincentmli** [comment]
> ```
> cilium install --version=v1.11.4 --kube-proxy-replacement strict --helm-set-string=k8sServiceHost=k8s-api.example.com,k8sServicePort=6443,ipv6.enabled=true,ipam.mode=kubernetes,enableIPv6Masquerade=false,externalIPs=true
> ```

@vosdev the cilium cli says `unknown flag: --helm-set-string`

```
cilium install --version=v1.12.0-rc2 --kube-proxy-replacement strict --helm-set-string=k8sServiceHost=10.169.72.9,k8sServicePort=6443
unknown flag: --helm-set-string
```
cilium cli version
```
cilium-cli: v0.10.5 compiled with go1.18.1 on linux/amd64
```

**Comment by vincentmli** [comment]
`cilium config  set k8sServicePort "6443"` and `cilium config set k8sServiceHost "10.169.72.9"` works after `cilium install`

**Comment by tklauser** [comment]
@vincentmli could you try cilium-cli v0.11.7? This should have the `--helm-set-string` flag.

**Comment by vincentmli** [comment]
> @vincentmli could you try cilium-cli v0.11.7? This should have the `--helm-set-string` flag.
cilium-cli v0.11.7 works, I thought 0.10.5 is most recent lease, missed 0.11.7 release

**Comment by tklauser** [comment]
> cilium-cli v0.11.7 works, I thought 0.10.5 is most recent lease, missed 0.11.7 release

cilium v0.10.5 was released more recently, thus unfortunately https://github.com/cilium/cilium-cli/releases/latest/ currently points to it. However, that version is only a maintainance release to support Cilium v1.10.x for which installation with the helm-based cilium-cli v0.11.x doesn't work. For installation of Cilium ≥v1.11.x the latest cilium-cli v0.11.7 should be used.

**Comment by tklauser** [comment]
> it can be configured through the new --set-helm-string option like such:
> 
> ```
> cilium install --version=v1.11.4 --kube-proxy-replacement strict --helm-set-string=k8sServiceHost=k8s-api.example.com,k8sServicePort=6443,ipv6.enabled=true,ipam.mode=kubernetes,enableIPv6Masquerade=false,externalIPs=true
> ```
> 
> I believe this issue can be closed as #717 has been merged some time ago.

I agree. The original issue was fixed by #717 which is part of the cilium-cli v0.11.x releases. Thus closing this issue.
